spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/couchbase/CouchbaseProperties.java

```diff
@@ -41,8 +48,28 @@
 
 	private final Env env = new Env();
 
+	private int bootstrapHttpDirectPort = DefaultCoreEnvironment.BOOTSTRAP_HTTP_DIRECT_PORT;
+
+	private int bootstrapCarrierDirectPort = DefaultCoreEnvironment.BOOTSTRAP_CARRIER_DIRECT_PORT;
+
+	public int getBootstrapHttpDirectPort() {
+		return this.bootstrapHttpDirectPort;
+	}
+
+	public void setBootstrapHttpDirectPort(int bootstrapHttpDirectPort) {
+		this.bootstrapHttpDirectPort = bootstrapHttpDirectPort;
+	}
+
+	public int getBootstrapCarrierDirectPort() {
+		return this.bootstrapCarrierDirectPort;
+	}
+
+	public void setBootstrapCarrierDirectPort(int bootstrapCarrierDirectPort) {
+		this.bootstrapCarrierDirectPort = bootstrapCarrierDirectPort;
+	}
+
 	public List<String> getBootstrapHosts() {
-		return this.bootstrapHosts;
+		return Objects.nonNull(this.bootstrapHosts) ? this.bootstrapHosts : DEFAULT_HOSTS;
```

Comment:
I don't think we should change that there. If the boostrap hosts aren't set, it should return `null` as before. Adding a logic of a "default" host while the property has no default value is wrong.

Reply:
When CouchbaseMock is initialized on a random available port, these 2 ports are also changed from defaults to available ones. For a couchbase-client to be able to connect to CouchbaseMock we need to set these properties to the ports on which CouchbaseMock is actually listening.

Reply:
Correct, forgot to remove it.
The idea was, to create the same behavior as for mongo:
https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mongo/MongoProperties.java#L46

So that you can start couchbase-based application without any properties when CouchbaseMock is on classpath